arangod/RestHandler/RestRepairHandler.cpp

```diff
@@ -0,0 +1,699 @@
+          "prototype shard, after they should already have been fixed. "
+          "This should not happen, but it should be safe to try this job "
+          "again. "
+          "If that does not help, please report this error.";
+      break;
+    case TRI_ERROR_CLUSTER_REPAIRS_INCONSISTENT_ATTRIBUTES:
+      errorDetails =
+          "Error while collecting repair actions. "
+          "Unexpected state of distributeShardsLike or "
+          "repairingDistributeShardsLike attribute. "
+          "This should not happen, but it should be safe to try this job "
+          "again. "
+          "If that does not help, please report this error.";
+      break;
+    case TRI_ERROR_CLUSTER_REPAIRS_MISMATCHING_SHARDS:
+      errorDetails =
+          "Error while collecting repair actions. "
+          "In this collection, some shard and its distributeShardsLike "
+          "prototype have an unequal number of DBServers. This has to be fixed "
+          "before this collection can be repaired.";
+      break;
+    case TRI_ERROR_CLUSTER_REPAIRS_JOB_FAILED:
+      errorDetails =
+          "Error during repairs! "
+          "Moving a shard failed. Did you do any changes to the affected "
+          "collection(s) or the cluster during the repairs? It should be safe "
+          "to try this job again. "
+          "If that does not help, please report this error.";
+      break;
+    case TRI_ERROR_CLUSTER_REPAIRS_JOB_DISAPPEARED:
+      errorDetails =
+          "Error during repairs! "
+          "A job to move a shard disappeared. This should not happen. "
+          "Please report this error.";
+      break;
+    case TRI_ERROR_CLUSTER_REPAIRS_OPERATION_FAILED:
+      errorDetails =
+          "Error during repairs! "
+          "Executing an operation as an agency transaction failed. Did you do "
+          "any changes to the affected collection(s) or the cluster during the "
+          "repairs? It should be safe to try this job again. "
+          "If that does not help, please report this error.";
+      break;
+    default:
+        // Some non-repair related error
+        ;
+  }
+
+  if (errorDetails.is_initialized()) {
+    builder.add("errorDetails", VPackValue(errorDetails.get()));
+  }
+}
+
+bool RestRepairHandler::pretendOnly() { return _pretendOnly; }
+
+ResultT<bool> RestRepairHandler::checkReplicationFactor(
+    DatabaseID const& databaseId, CollectionID const& collectionId) {
+  ClusterInfo* clusterInfo = ClusterInfo::instance();
+  if (clusterInfo == nullptr) {
+    LOG_TOPIC(ERR, arangodb::Logger::CLUSTER)
+        << "RestRepairHandler::checkReplicationFactor: "
+        << "No ClusterInfo instance";
+    generateError(rest::ResponseCode::SERVER_ERROR,
+                  TRI_ERROR_HTTP_SERVER_ERROR);
+
+    return Result(TRI_ERROR_INTERNAL);
+  }
+
+  clusterInfo->loadPlan();
+  std::shared_ptr<LogicalCollection> const collection =
+      clusterInfo->getCollection(databaseId, collectionId);
```

Comment:
Not necessarily. If `_planProt.isValid` it might return an old version as far as I can see: https://github.com/arangodb/arangodb/blob/26842ae83b63397a6f400f2e68ce940d9434b00b/arangod/Cluster/ClusterInfo.cpp#L907